parse_go1.23_test.go

```diff
@@ -0,0 +1,45 @@
+//go:build go1.23
```

Comment:
Considering the build matrix in `ci.yml` would this file every be executed during our builds?

Reply:
Sure but maybe expand the matrix to include these later Go versions that we are interested in?

Reply:
Maybe just add Go 1.23 then? Or it will break the build if you add it?

Reply:
I'll try, but my guess is it breaks, yes

Reply:
🎉